lib/crypto.js

```diff
@@ -282,7 +282,16 @@ Sign.prototype.sign = function sign(options, encoding) {
 
   var key = options.key || options;
   var passphrase = options.passphrase || null;
-  var ret = this._handle.sign(toBuf(key), null, passphrase, options);
+
+  // Options specific to RSA
+  var rsaPadding = options.padding || constants.RSA_PKCS1_PADDING;
+  var pssSaltLength = constants.RSA_PSS_SALTLEN_MAX_SIGN;
+  if (typeof options.saltLength === 'number') {
+    pssSaltLength = options.saltLength;
+  }
```

Comment:
Since negative numbers don't make sense, I'd check that `x === x >>> 0` (logical shift) and switch the C++ code to IsUint32() and ToUint32().

If you want to stick with signed numbers, check `x === x >> 0` (arithmetic shift) instead.